csh_auth.go

```diff
@@ -0,0 +1,156 @@
+package csh_auth
+
+import (
+    "fmt"
+    "os"
+    "net/http"
+    "time"
+    log "github.com/sirupsen/logrus"
+    jwt "github.com/dgrijalva/jwt-go"
+    oidc "github.com/coreos/go-oidc"
+    "golang.org/x/net/context"
+    "golang.org/x/oauth2"
+    "github.com/gin-gonic/gin"
+)
+
+
+var clientID = os.Getenv("csh_auth_client_id")
+var clientSecret = os.Getenv("csh_auth_client_secret")
+var secret = os.Getenv("csh_auth_jwt_secret")
+var state = os.Getenv("csh_auth_state")
+var server_host = os.Getenv("csh_auth_server_host")
+var redirect_uri = os.Getenv("csh_auth_redirect_uri")
+
+var AuthKey = "cshauth"
+var provider_uri = "https://sso.csh.rit.edu/realms/csh"
+
+var authenticate_uri = "/authenticate"
+// globals eww
+var config oauth2.Config // this guy changes a bit, weird
+var ctx context.Context
```

Comment:
contexts shouldn't be long lived like this. They are usually scoped to the lifetime of a single request or a background task.